application/src/main/java/org/thingsboard/server/service/security/auth/oauth2/AbstractOAuth2ClientMapper.java

```diff
@@ -206,9 +206,10 @@ private CustomerId getCustomerId(TenantId tenantId, String customerName) {
     }
 
     private Optional<DashboardId> getDashboardId(TenantId tenantId, String dashboardName) {
-        PageLink searchTextLink = new PageLink(1, 0, dashboardName);
+        PageLink searchTextLink = new PageLink(Integer.MAX_VALUE, 0, dashboardName);
```

Comment:
This fix is far from the best possible solution. Please add one more method to the DashboardInfoRepository to find the first (or all )dashboards with the exact name instead of search text.

Reply:
In the PE version, it's going to be a code duplicate due to the method that searches by groupsIds.
Can we consider refactoring this code snippet to correct pagination by 100 dashboards? 
PageLink searchTextLink = new PageLink(100, 0, dashboardName);
do {
...
} while()